src/com/projects/Performer.java

```diff
@@ -0,0 +1,47 @@
+package com.projects;
+
+import java.util.List;
+
+public class Performer extends Thread {
+    private final StorageManager storageManager;
+    private final int searchNumber;
+
+    private volatile boolean startSearch = false;
+
+    private List<Integer> indexes;
```

Comment:
why this list is field but not local variable?

Reply:
fixed